system/Database/BaseBuilder.php

```diff
@@ -2171,94 +2283,29 @@ protected function validateUpdate(): bool
     }
 
     /**
-     * Compiles an update string and runs the query
+     * Sets data and calls batchExecute to run queryies
      *
-     * @return false|int|string[] Number of rows affected or FALSE on failure, SQL array when testMode
+     * @param array|object|string|null $set         a dataset or select query
+     * @param array|RawSql|string|null $constraints
      *
-     * @throws DatabaseException
+     * @return false|int|string[] Number of rows affected or FALSE on failure, SQL array when testMode
      */
-    public function updateBatch(?array $set = null, ?string $index = null, int $batchSize = 100)
+    public function updateBatch($set = null, $constraints = null, int $batchSize = 100)
```

Comment:
Changed:
1. type of `$set`
2. type of `$index`
3. name of `$index`

Please write all of them in the changelog.

Reply:
This has already been written in change log. I did add the change of type of $set though.